src/payment/payment.ts

```diff
@@ -80,6 +80,11 @@ export interface PaypalInstrument {
     };
 }
 
+export interface BarclaycardInstrument {
```

Comment:
Not sure why this payload needs to tie with a specific payment method. It seems they are just `HostedInstrument` and `VaultedInstrument` in a different format. I believe moving forward we want to use `FormattedPayload<T>`. So perhaps you could just create two new interfaces and call them `FormattedHostedInstrument` and `FormattedVaultedInstrument`?